include/zephyr/usb/msos_desc.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief MS OS 2.0 descriptor definitions
+ *
+ */
+
+#include <stdint.h>
+
+enum ms_os_20_descriptor_type {
+	MS_OS_20_SET_HEADER_DESCRIPTOR		= 0x00,
+	MS_OS_20_SUBSET_HEADER_CONFIGURATION	= 0x01,
+	MS_OS_20_SUBSET_HEADER_FUNCTION		= 0x02,
+	MS_OS_20_FEATURE_COMPATIBLE_ID		= 0x03,
+	MS_OS_20_FEATURE_REG_PROPERTY		= 0x04,
+	MS_OS_20_FEATURE_MIN_RESUME_TIME	= 0x05,
+	MS_OS_20_FEATURE_MODEL_ID		= 0x06,
+	MS_OS_20_FEATURE_CCGP_DEVICE		= 0x07,
+	MS_OS_20_FEATURE_VENDOR_REVISION	= 0x08
+};
+
+enum ms_os_20_property_data_type {
+	MS_OS_20_PROPERTY_DATA_RESERVED			= 0,
+	MS_OS_20_PROPERTY_DATA_REG_SZ			= 1,
+	MS_OS_20_PROPERTY_DATA_REG_EXPAND_SZ		= 2,
+	MS_OS_20_PROPERTY_DATA_REG_BINARY		= 3,
+	MS_OS_20_PROPERTY_DATA_REG_DWORD_LITTLE_ENDIAN	= 4,
+	MS_OS_20_PROPERTY_DATA_REG_DWORD_BIG_ENDIAN	= 5,
+	MS_OS_20_PROPERTY_DATA_REG_LINK			= 6,
+	MS_OS_20_PROPERTY_DATA_REG_MULTI_SZ		= 7
+};
+
+/* Microsoft OS 2.0 descriptor set header */
+struct ms_os_20_root_desc {
```

Comment:
okay, my fault, I looked at the spec and what we have in the tree.
I think we can omit `_desc` from all of them because it is clear what it is and and it will not conflict with anything else. We already use msosv1 and msosv2 prefixes in the tree, so I suggest we follow that and name it:
```c
struct msosv2_descriptor_set_header {};
...
struct msosv2_configuration_subset_header{};
...
struct msosv2_function_subset_header{};
...
struct msosv2_compatible_id{};
...
```